services/eth_gateway.go

```diff
@@ -0,0 +1,69 @@
+package services
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+// TODO: Find etherum node to connect to.
+const ethUrl = "ws://0.0.0.0"
+
+// Singleton client
+var (
+	client *ethclient.Client
+	mtx    sync.Mutex
+)
+
+func sharedClient() (c *ethclient.Client, err error) {
+	if client != nil {
+		return client, nil
+	}
+
+	// check-lock-check pattern to avoid excessive locking.
+	mtx.Lock()
+	defer mtx.Unlock()
+
+	if client != nil {
+		c, err = ethclient.Dial(ethUrl)
+		if err != nil {
+			fmt.Println("Failed to connect to Etherum node.")
+			return
+		}
+
+		// Sets Singleton
+		client = c
+	}
+
+	return client, err
+}
+
+func BuryPrl() {
```

Comment:
Will this method be creating the addresses and getting the private seed keys for the eth addresses we are going to use for the treasure?

Reply:
I thought this was a function in the smart contract called from the eth address of the upload session?

I will have to add another function to generate a new eth address for an upload session though.

Reply:
There's definitely a bury function but I'm not sure that it creates those addresses.